CONTRIBUTING.md

```diff
@@ -142,3 +142,86 @@ sense for any commit, e.g.
     Acked-by:
     Sponsored-by:
 ```
+
+## Pull requests
+
+Pull requests with patches, improvements, new features are a great help.
+Please keep them clean from unwanted commits.
+
+Follow the steps to get your work included in the project.
+
+1. [Fork](https://help.github.com/fork-a-repo/) the project, clone your fork,
+   and add the fossology remote:
+
+   ```bash
+   # Clone your fork of the repo into the current directory
+   git clone https://github.com/<your-username>/fossology.git
+   # Navigate to the cloned directory
+   cd fossology
+   # Assign the original repo to a remote called "upstream"
+   git remote add upstream https://github.com/fossology/fossology.git
+   ```
+
+2. Get the latest changes from upstream:
+
+   ```bash
+   git checkout master
+   git pull upstream master
+   ```
+
+3. Create a new branch from the main master branch to contain your changes.
+   Best way is to call is to follow the type described in **Git Commit Conventions**
+   stated above: `<githubId>/#<issueNr>/<description/scope/topic>`
+
+   ```bash
+   git checkout -b <topic-branch-name>
+   ```
+
+   Example:
+
+   ```bash
+   git checkout -b john/issue-###/buckets-undefined-index
```

Comment:
wouldn't it be correct to write `john/#xxx/buckets-undefined-index`

Reply:
I am fine with anything. Is it okay now?